numpy/core/memmap.py

```diff
@@ -79,7 +79,7 @@ class memmap(ndarray):
         of remaining bytes after `offset` is not a multiple of the byte-size
         of `dtype`, you must specify `shape`. By default, the returned array
         will be 1-D with the number of elements determined by file size
-        and data-type.
+        and data-type. If ``mode == 'w+'`` then shape must also be specified.
```

Comment:
Just a nit, but it might make more sense for this sentence to be added to the description of the `mode` parameter.

Reply:
Thank you for the feedback! I will fix this. 

This is my first proper pull request. 

One of the tests has failed (Test Emscripten/Pyodid build) however I don't understand the error nor how to fix it. Could you enlighten me perhaps?

Reply:
I have now updated  this change, moving this statement to line 62.